torchvision/datasets/sun397.py

```diff
@@ -0,0 +1,95 @@
+from pathlib import Path
+from typing import Any, Tuple, Callable, Optional
+
+import PIL.Image
+
+from .utils import verify_str_arg, download_and_extract_archive
+from .vision import VisionDataset
+
+
+class SUN397(VisionDataset):
+    """`The SUN397 Data Set <https://vision.princeton.edu/projects/2010/SUN/>`_.
+
+    The SUN397 or Scene UNderstanding (SUN) is a dataset for scene recognition consisting of
+    397 categories with 108'754 images. The dataset also provides 10 partitions for training
+    and testing, with each partition consisting of 50 images per class.
+
+    Args:
+        root (string): Root directory of the dataset.
+        split (string, optional): The dataset split, supports ``"train"`` (default) and ``"test"``.
+        partition (integer, optional): A valid partition can be an integer from 1 (default) to 10 or None,
+            for the entire dataset.
```

Comment:
@pmeier it looks like this `partition` parameter has a different meaning from that of DTD, where a given "partition" covers the entire dataset, not a subset of it. If that's the case, we should think of a different name (either for this one or for that of DTD).

Reply:
yes in the case of SUN397, partition is a random subset of whole dataset with 50 images per class. Let me know if changes needs to be made to the naming, I will do this in this PR itself. @pmeier

Reply:
We've discussed this a bit with @pmeier , let's keep it as it is here. We might change that of DTD but we'll do that later. Thanks for checking @saswatpp

Reply:
We should stick with `partition` here and change the name in `DTD`.

Reply:
Ok Sure <3